solvers/NonlinearProgram.m

```diff
@@ -155,7 +155,6 @@
       obj.bbcon_lb = [];
       obj.bbcon_ub = [];
       
-      obj = obj.setSolver('default');
```

Comment:
With this change we'd need to set the solver to 'default' in the constructor of every class that inherits from `NonlinearProgram`. What's your motivation for this change?